### ISSUE TITLE
New smartthings app

### ISSUE THREAD

**Issue by jasonsf** [nutechsoftware/alarmdecoder-smartthings#55]
Are there any thought/plans to support the new smartthings app? Specifically some kind of UI for showing zone status like the old one?

**Comment by djkinley** [comment]
Any update on this?

**Comment by f34rdotcom** [comment]
Yes. Progress is good. I have a few issues I am working on with ST engineers but I am mostly functional. I still need to add zones but as soon as they figure this one issue I am having then I will work on it next. I am testing with a 10$ ESP32 dev kit board but long term it will be an Ethernet + Wifi based ESP32 board.

https://github.com/nutechsoftware/AlarmDecoder-STSDK

**Comment by jasonsf** [comment]
Can you explain a little further what this will be? I was expecting a smart
app for this, but it sounds like you are putting together new hardware. How
does this new device fit physically in a network? I currently have a
raspberry pi with your ad2pi connected to my alarm system. Will that need
to be replaced to use this new board? Or will it work in conjunction with
it?

Jason

**Comment by f34rdotcom** [comment]
Ya its a bit confusing but it is the easy path into the SmartThings platform. 

You will be able to use your Pi and just get a 10$ ESP32 board to program with this code and be back in business with the new ST App. You can then later use your existing Pi board and build the new appliance with it if you want. I plan on selling upgrade kits with case and parts.

This project is just open source code using the SDK from SmartThings to make a direct attached device on the ST cloud. This will show up on your new ST App and can be adopted using your phone. Very plugnplay as it uses Wifi to find and connect the device to the ST cloud. The ESP32 board can connect to your existing AD2Pi board over ser2sock or directly to an AD2Pi or pHAT board

The new appliance will look roughly like this.
![Test assembly AD2IOT v6](https://user-images.githubusercontent.com/2042644/97068075-199edd80-1579-11eb-87d2-2414840bad84.png)

Hope this helps let me know if you have questions or suggestions.

Best
 SM

The new ST app has a much more simple interface but hopefully it will grow some more advanced UI features over time.
For now the basics will be covered.
![Screenshot_20201019-122957_SmartThings](https://user-images.githubusercontent.com/2042644/97068101-579c0180-1579-11eb-9298-254195563536.jpg)

**Comment by macbentosh** [comment]
Can I ask why hardware will be required to restore functionality previously had?

**Comment by macbentosh** [comment]
Any update on this?

**Comment by dreadirester** [comment]
Any Updates on this?

**Comment by f34rdotcom** [comment]
Yes the code is up and running works very well. Future roadmap includes custom UI when the dust settles on the new platform with Samsung SmartThings. https://github.com/nutechsoftware/AlarmDecoder-STSDK for more info or support refer to that project. I will be keeping this project but renaming it for Hubitat where it is still working.

Until my device is "certified" by Samsung it will still only be available if you join the Nu Tech Software dev group on samsung dev site and eanble developer mode on the new ST App as per the README.MD document.

I will start selling the new appliance soon I am expecting the first batch of 50 ESP32-POE-ISO boards here in the next few weeks and I have 10 carrier assemblies ready to ship when they do. The new device is on my store but not enabled yet but will by the end of March or sooner. Still no cases as I have not received the equipment I need to CNC out the holes for the ethernet and other things.

Best
 Sean Mathews.

**Comment by f34rdotcom** [comment]
> Can I ask why hardware will be required to restore functionality previously had?
Sure sorry for the very late reply was busing wring a lot of new code.

I hope to be able to run the same code on a Pi maybe using an emulator but I would prefer bare metal code. I did submit a request and it was accepted as a feature request by the Samsung dev team. We shall see. In general though the Pi is overkill for the task and especially with the Pi4 it has gotten even worse. The ESP32 is much less heat, power, $.
https://github.com/SmartThingsCommunity/st-device-sdk-c-ref/issues/39

For now people with a Pi and AD2Pi can just run ser2sock on the Pi and configure most any $9 ESP32 board with this firmware and have it connect to the AD2 over the network. This puts all of the STSDK logic in the ESP32 and your Pi is just a gateway to the AD2* for it to use.

Smartthings/Samsung have only developed this API for "embedded IoT" hardware and that includes currently ESP32 boards.
https://smartthings.developer.samsung.com/docs/devices/direct-connected-devices/setup-environment.html

Best
 Sean M.
